### DISCUSSION TITLE
Nutale and presence

### DISCUSSION THREAD

**Discussion by Tonio16** [theengs/gateway#181] [General]
Hello

I am a new user of theengs gateway and Bluetooth devices for my home automation.

I bought some Nutale devices during Black Friday:
https://www.amazon.fr/Nutale-Localisateur-Portefeuilles-Anti-Perte-Trouve-cl%C3%A9s/dp/B08HM2D4W3

The first one that I activated is apperaring in Theengs default topic as:
`{"servicedatauuid": "0900", "servicedata": "a4c1381xxyyzz160100010100", "name": "nutale", "id": "A4:C1:38:xx:yy:zz", "rssi": -48}`

I don't see him in the presence topic. What shoud I do to have them in?

Here it's my docker-compose set-up:
> gateway:
    image: theengs/gateway
    network_mode: host
    container_name: theengs-gateway
    environment:
       - MQTT_HOST=192.168.x.y
       - MQTT_USERNAME=<jeedom>
       - MQTT_PASSWORD=<jeedom>
       - MQTT_PUB_TOPIC=home/docker/BTtoMQTT #
       - MQTT_SUB_TOPIC=home/+/BTtoMQTT/undecoded #
       - MQTT_PRE_TOPIC=home/presence/TheengsGateway #
       - LWT_TOPIC=home/docker/LWT
       - PRESENCE=true #
       - PUBLISH_ALL=true #
       - TIME_BETWEEN=60
       - SCAN_DUR=10 #
       - SCAN_TIME=60
       - LOG_LEVEL=ERROR
       - HASS_DISCOVERY=true
       - DISCOVERY=true
       - DISCOVERY_TOPIC=homeassistant/sensor
       - DISCOVERY_DEVICE_NAME=docker
       - DISCOVERY_FILTER=[IBEACON,GAEN,MS-CDP,APPLE_CONT,APPLE_CONTAT]  #
       - SCANNING_MODE=active
       - ADAPTER=hci0
    labels:
       - "com.centurylinklabs.watchtower.enable=true"
    restart: unless-stopped
    volumes:
       - /var/run/dbus:/var/run/dbus

Thanks for your help

Antoine

**Comment by DigiH** [comment]
Hi @Tonio16 

Is your above posted MQTT message with raw **servicedatauuid** and **servicedata** the only one you are receiving for your Nutale device, or might there be other ones as well, possibly with **manufacturerdata**?

This can be best seen in MQTT Explorer's message history of the device.

With the Nutale not being included and recognised yet, it won't produce any presence messages, but with your information and possibly some additional **manufacturerdata** messages as well it can be included in a future update.

Thanks

**Comment by Tonio16** [comment]
Hello 
Thanks for your answer. I started MQTT Explorer 2hours ago but I don't see any topic with manufacturerdata for the nutale.
I linked them to the app of nutale. Could it be the reason?
I tried with an upaired one, it doesn't so much stuff, except the service data part:
- 160100010100 for paired one
- 160100010000 for unpaired one
`{"servicedatauuid": "0900", "servicedata": "a4c138xxyyzz160100010000", "name": "nutale", "id": "A4:C1:38:xx:yy:zz", "rssi": -76}`

Antoine

**Comment by DigiH** [comment]
Thank you for the confirmation @Tonio16 

I have prepared a decoder for merging into Theengs Decodert, which then will be included inn the next release of Theengs Gateway

https://github.com/theengs/decoder/pull/457

which I will merge now after your response. After the inclusion in Theengs Gateway you should then also see presence messages for the nutale.

**Reply by Tonio16** [reply]
Thanks a lot!

Antoine

**Reply by Tonio16** [reply]
Hello

I notice today that one as a different service data ending.
`{"servicedatauuid": "0900", "servicedata": "a4c138xxyyz160100010101", "name": "nutale", "id": "A4:C1:38:xx:yy:zz", "rssi": -77}`

So we have three ending:
160100010101
160100010000
160100010100

Why? I don't know.

Antoine

**Reply by DigiH** [reply]
Hi, that's interesting …

**1.** It could be a low battery indicator - does the battery show low in the native app?
**2.** Does this nutale have a button and was it pressed, on purpose or accidentally, to flag this indicator?

As you mentioned before

> * 160100010**1**00 for paired one
> * 160100010**0**00 for unpaired one

the 3rd from last 1 indicates a paired nutale. So with your last observation
> 160100010101

it was also paired.
**3.** So it could also mean that the nutale was out of reach of the paired phone.

Let us know if you find out what this trailing 1 might mean so we might include it in the decoder.

**Reply by Tonio16** [reply]
Hello
.the paired unpaired assumption from me,  is probably wrong. I did it based on the paired one vs the unpaired. 

I paired now the unpaired one, and the number is the same, so no change before and after pairing.
But as you suggested,  I would investigate further.

Antoine

**Comment by Tonio16** [comment]
Hello

I used the same nutale for the test with only one gateway:
Unpaired Nutale:
`{"servicedatauuid": "0900", "servicedata": "a4c138xxyyzz160100010000", "name": "nutale", "id": "A4:C1:38:xx:yy:zz", "rssi": -45}`

After pairing
`{"servicedatauuid": "0900", "servicedata": "a4c138xxyyzz160100010100", "name": "nutale", "id": "A4:C1:38:xx:yy:zz", "rssi": -50}`

I see the 101 again when I stopped bluetooth of my phone with the app.
`{"servicedatauuid": "0900", "servicedata": "a4c138xxyyzz160100010101", "name": "nutale", "id": "A4:C1:xx:yy:zz", "rssi": -49}`

Moreover, the nutale are probably not seen when the app is activated and they are "communicating" together, since I don't have updates of the topic.

I set a nutale in pairing mode close to the gateway but no change in theengs and so in the servicedata part was catched.

Cheers

Antoine

**Reply by DigiH** [reply]
Thanks for the update with your further investigations.

I suppose my assumption 3. was correct then - the 3rd from last 1 indicates the out of reach of the paired phone.

I'm not sure this could be helpful much by defining it in a separate decoder for this, similar to the 5th from last paired/unpaired, so my suggestion is to leave the nutale tracker implementation as it currently is. Unless you and other nutale users strongly disagree ;)
